src/main/fc/fc_tasks.c

```diff
@@ -38,6 +38,13 @@
 #include "drivers/stack_check.h"
 #include "drivers/transponder_ir.h"
 #include "drivers/vtx_common.h"
+#ifdef USB_CDC_HID
+//TODO: Make it platform independent in the future
+#include "vcpf4/usbd_cdc_vcp.h"
+#include "usbd_hid_core.h"
+//TODO: Nicer way to handle this...
+#undef MIN
```

Comment:
What is this line needed for? This seems to be a very risky thing to do, as `MIN` is a commonly used define.

Reply:
compiler complains, MIN is defined in 2 headers. It's included in one of the next headers anyway (imu.h) so this doesn't break anything

Reply:
In terms of maintainability, and to avoid hard to trace down build problems when target specific defines are used I'd prefer not to rely on this being defined later on.

Also, if `#pragma once` is used consistently (as it should), the later include will not define this again.

I can look into this later on.

Reply:
It's not that we have same nested maths.h included in two headers. usb_defines.h declares it's own MIN. I have another way to handle this if you wan't to have a look.

Reply:
My suggestion:

```
diff --git a/lib/main/STM32_USB_Device_Library/Class/hid/src/usbd_hid_core.c b/lib/main/STM32_USB_Device_Library/Class/hid/src/usbd_hid_core.c
index 9b5b6246d..15dd2bda8 100644
--- a/lib/main/STM32_USB_Device_Library/Class/hid/src/usbd_hid_core.c
+++ b/lib/main/STM32_USB_Device_Library/Class/hid/src/usbd_hid_core.c
@@ -51,6 +51,7 @@
 #include "usbd_desc.h"
 #include "usbd_req.h"
 
+#include "common/maths.h"
 #include "common/utils.h"
 
 
diff --git a/lib/main/STM32_USB_Device_Library/Class/msc/src/usbd_msc_bot.c b/lib/main/STM32_USB_Device_Library/Class/msc/src/usbd_msc_bot.c
index 4f53a4f25..40557abb5 100644
--- a/lib/main/STM32_USB_Device_Library/Class/msc/src/usbd_msc_bot.c
+++ b/lib/main/STM32_USB_Device_Library/Class/msc/src/usbd_msc_bot.c
@@ -31,6 +31,8 @@
 #include "usbd_ioreq.h"
 #include "usbd_msc_mem.h"
 
+#include "common/maths.h"
+
 #define UNUSED(x) (void)(x)
 
 /** @addtogroup STM32_USB_OTG_DEVICE_LIBRARY
diff --git a/lib/main/STM32_USB_Device_Library/Class/msc/src/usbd_msc_scsi.c b/lib/main/STM32_USB_Device_Library/Class/msc/src/usbd_msc_scsi.c
index bd50a973f..c8047bf00 100644
--- a/lib/main/STM32_USB_Device_Library/Class/msc/src/usbd_msc_scsi.c
+++ b/lib/main/STM32_USB_Device_Library/Class/msc/src/usbd_msc_scsi.c
@@ -31,6 +31,7 @@
 #include "usbd_msc_mem.h"
 #include "usbd_msc_data.h"
 
+#define  MIN(a, b)      (((a) < (b)) ? (a) : (b))
 #define UNUSED(x) (void)(x)
 
 
diff --git a/lib/main/STM32_USB_Device_Library/Core/src/usbd_core.c b/lib/main/STM32_USB_Device_Library/Core/src/usbd_core.c
index 0613f5018..0b9f03a77 100644
--- a/lib/main/STM32_USB_Device_Library/Core/src/usbd_core.c
+++ b/lib/main/STM32_USB_Device_Library/Core/src/usbd_core.c
@@ -32,6 +32,8 @@
 #include "usb_dcd_int.h"
 #include "usb_bsp.h"
 
+#include "common/maths.h"
+
 /** @addtogroup STM32_USB_OTG_DEVICE_LIBRARY
 * @{
 */
diff --git a/lib/main/STM32_USB_Device_Library/Core/src/usbd_req.c b/lib/main/STM32_USB_Device_Library/Core/src/usbd_req.c
index b74e82e3a..d9abef4fc 100644
--- a/lib/main/STM32_USB_Device_Library/Core/src/usbd_req.c
+++ b/lib/main/STM32_USB_Device_Library/Core/src/usbd_req.c
@@ -30,6 +30,7 @@
 #include "usbd_ioreq.h"
 #include "usbd_desc.h"
 
+#include "common/maths.h"
 
 /** @addtogroup STM32_USB_OTG_DEVICE_LIBRARY
   * @{
diff --git a/lib/main/STM32_USB_OTG_Driver/inc/usb_defines.h b/lib/main/STM32_USB_OTG_Driver/inc/usb_defines.h
index 34cc8520c..2e8d39929 100644
--- a/lib/main/STM32_USB_OTG_Driver/inc/usb_defines.h
+++ b/lib/main/STM32_USB_OTG_Driver/inc/usb_defines.h
@@ -174,8 +174,6 @@
 #define HCCHAR_BULK                            2
 #define HCCHAR_INTR                            3
 
-#define  MIN(a, b)      (((a) < (b)) ? (a) : (b))
-
 /**
   * @}
   */
diff --git a/src/main/fc/fc_tasks.c b/src/main/fc/fc_tasks.c
index 0b342ebed..f8546576a 100644
--- a/src/main/fc/fc_tasks.c
+++ b/src/main/fc/fc_tasks.c
@@ -42,8 +42,6 @@
 //TODO: Make it platform independent in the future
 #include "vcpf4/usbd_cdc_vcp.h"
 #include "usbd_hid_core.h"
-//TODO: Nicer way to handle this...
-#undef MIN
 #endif
 
 #include "fc/config.h"
```

Alternatively, we can add the define `#define  MIN(a, b)      (((a) < (b)) ? (a) : (b))` to all library sources that need it.